Apps/WebClient/src/ClientApp/src/store/modules/credential/actions.ts

```diff
@@ -0,0 +1,92 @@
+import { ResultError } from "@/models/requestResult";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.container";
+import { ICredentialService, ILogger } from "@/services/interfaces";
+
+import { CredentialActions } from "./types";
+
+export const actions: CredentialActions = {
+    retrieveConnection(context, params: { hdid: string }): Promise<boolean> {
+        const logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+        const credentialService: ICredentialService =
+            container.get<ICredentialService>(
+                SERVICE_IDENTIFIER.CredentialService
+            );
+        return new Promise((resolve, reject) => {
+            if (context.getters.patientData.hdid !== undefined) {
+                logger.debug(
+                    `Credential Connection found stored, not querying!`
+                );
```

Comment:
I don't follow how this condition catches the case where the connection already exists.

Reply:
Right, I meant the check on `patientData.hdid`.  But I see you've addressed that in the latest commit.